src/api/RestAPI.ts

```diff
@@ -196,27 +197,24 @@ export class RestAPI extends EventEmitter {
 
     private _jwtRaw: string = null;
 
-    protected handleUpdateJWT: (jwt: string) => void = () => undefined;
+    protected handleUpdateJWT: (jwt: string) => void;
 
     constructor(options: RestAPIOptions = {}) {
         super();
 
         this.endpoint = options.endpoint || process.env[ENV_KEY_ENDPOINT] || DEFAULT_ENDPOINT;
         this.origin = options.origin || this.origin;
         this.jwtRaw = options.jwt || process.env[ENV_KEY_APPLICATION_JWT];
-
-        if (options.handleUpdateJWT && typeof options.handleUpdateJWT === "function") {
-            this.handleUpdateJWT = options.handleUpdateJWT;
-        }
+        this.handleUpdateJWT = options.handleUpdateJWT || undefined;
 
         this.appId = options.appId || process.env[ENV_KEY_APP_ID];
         this.secret = options.secret || process.env[ENV_KEY_SECRET];
         this.authToken = options.authToken;
     }
 
     set jwtRaw(jwtRaw: string) {
-        this.jwt = parseJWT(jwtRaw);
         this._jwtRaw = jwtRaw;
+        this.jwt = parseJWT(jwtRaw);
```

Comment:
This is the change... Although this is supposed to be synchronous after thinking about it... Really not sure what went wrong...